### PR TITLE
feat: add support for MMR to the MerkleStore

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

This adds support to importing the nodes of an MMR to the MerkleStore. The goal is to make testing of the Masm implementation easier to test (and this may be useful to initialize the store for the VM too).

review/merge after #117 

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/merkle/mmr/full.rs [hunk 1]
```diff
@@ -246,6 +255,108 @@ where
...
+pub enum Node {
+    Parent {
+        value: Word,
+        left: Word,
+        right: Word,
+    },
+    Leaf {
+        value: Word,
+    },
+}
```
Comment (bobbinth) [inline src/merkle/mmr/full.rs:271]: Similar comment to https://github.com/0xPolygonMiden/crypto/pull/117#discussion_r1154804910.

**Review by bobbinth — DISMISSED** [review]
Looks good! Thank you! Similar comment to the one I left in #117 - we could probably change this to return only internal nodes for now and that'll be sufficient for the `MerkleStore`.

File: src/merkle/mmr/full.rs (hunk 1 above)
Comment (hackaugusto) [reply src/merkle/mmr/full.rs]: rebased this PR on #117 , now it uses `InnerNodeInfo`.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**